### PR TITLE
feat(dup_enhancement#2): add more follower's info for master meta

### PR DESCRIPTION
**Pull request by Shuo-Jia**
# Ref-Issue
https://github.com/apache/incubator-pegasus/issues/892

# Change
This PR add more  follower's info for master meta, which will be used for master connect to follower

### CODE REVIEW

File: include/dsn/dist/replication/duplication_common.h [hunk 1]
```diff
@@ -70,5 +70,12 @@ inline bool is_cluster_id_configured(uint8_t cid)
     return get_distinct_cluster_id_set().find(cid) != get_distinct_cluster_id_set().end();
 }
 
+// todo(jiashuo1) it will be filled more constants
+class duplication_constants
```
Comment (levy5307) [inline include/dsn/dist/replication/duplication_common.h:74]: class ==> struct

File: src/meta/duplication/duplication_info.h [hunk 2]
```diff
@@ -42,13 +42,17 @@ class duplication_info
     /// \see duplication_info::decode_from_blob
     duplication_info(dupid_t dupid,
                      int32_t appid,
+                     std::string app_name,
                      int32_t partition_count,
                      uint64_t create_now_ms,
-                     std::string remote_cluster_name,
+                     std::string follower_cluster_name,
+                     std::vector<rpc_address> follower_cluster_metas,
```
Comment (levy5307) [inline src/meta/duplication/duplication_info.h:49]: use right value:
```
 std::vector<rpc_address> &&follower_cluster_metas,
```

File: src/meta/duplication/meta_duplication_service.cpp [hunk 3]
```diff
@@ -149,15 +149,24 @@ void meta_duplication_service::add_duplication(duplication_add_rpc rpc)
...
         response.__set_hint("failed to find cluster address in config [pegasus.clusters]");
         return;
     }
 
+    std::vector<rpc_address> meta_list;
+    dsn::replication::replica_helper::load_meta_servers(
+        meta_list,
+        duplication_constants::kClustersSectionKey.c_str(),
+        request.remote_cluster_name.c_str());
+
```
Comment (levy5307) [inline src/meta/duplication/meta_duplication_service.cpp:169]:
```
    std::vector<rpc_address> meta_list;
    dsn::replication::replica_helper::load_meta_servers(
        meta_list,
        duplication_constants::kClustersSectionKey.c_str(),
        request.remote_cluster_name.c_str());
    if (meta_list.empty()) {
        response.err = ERR_INVALID_PARAMETERS;
        response.__set_hint("failed to find cluster address in config [pegasus.clusters]");
        return;
    }
```

File: include/dsn/dist/replication/duplication_common.h (hunk 1 above)
Comment (hycdong) [reply include/dsn/dist/replication/duplication_common.h:74]: +1

File: src/meta/duplication/meta_duplication_service.cpp (hunk 3 above)
Comment (Shuo-Jia) [reply src/meta/duplication/meta_duplication_service.cpp:169]: the function `load_meta_servers` will core-dump if can't find, so add judge is redundant.

I still consider whether change the `load_meta_servers` action to `no core-dump`

Comment (levy5307) [reply src/meta/duplication/meta_duplication_service.cpp:169]: Changing the bahavior of `load_meta_servers` looks better to me :)

Comment (Shuo-Jia) [reply src/meta/duplication/meta_duplication_service.cpp:169]: Ok, but this is a common function, I will put other pr

Comment (levy5307) [reply src/meta/duplication/meta_duplication_service.cpp:169]: OK. Look forward to you next pull request.

**Review by levy5307 — APPROVED** [review]

**Review by hycdong — APPROVED** [review]

**Pull request merged**